xquant/keras/keras_report_utils.py

```diff
@@ -0,0 +1,380 @@
+        representative and validation datasets.
+
+        Args:
+            dataset (Any): The dataset to wrap.
+            is_validation (bool): Flag indicating if this is a validation dataset.
+            device (str, optional): Device to use for inference. Defaults to None.
+
+        Yields:
+            Any: Processed data from the dataset.
+        """
+
+        def process_data(x: Any, is_validation: bool) -> Any:
+            return x[0] if is_validation else x
+
+        for x in dataset():
+            yield process_data(x, is_validation)
+
+    def get_float_to_quantized_compare_points(self,
+                                              quantized_model: keras.Model,
+                                              float_model: keras.Model) -> Dict[str, str]:
+        """
+        Get comparison points between the floating-point and quantized models.
+
+        Args:
+            quantized_model (keras.Model): The quantized model.
+            float_model (keras.Model): The floating-point model.
+
+        Returns:
+            Dict[str, str]: A dictionary mapping comparison points between the two models.
+        """
+        quant_points_names = [
+            layer.name for layer in quantized_model.layers
+            if isinstance(layer, KerasQuantizationWrapper)
+        ]
+
+        float_name2quant_name = {}
+
+        for quant_point in quant_points_names:
+            candidate_float_layer_name = quantized_model.get_layer(quant_point).layer.name
+
+            if candidate_float_layer_name in [layer.name for layer in float_model.layers]:
+                if candidate_float_layer_name not in float_name2quant_name:
+                    float_name2quant_name[candidate_float_layer_name] = quant_point
+                else:
+                    Logger.critical(f"Duplicate mapping found for layer: {candidate_float_layer_name}")
+            else:
+                Logger.warning(f"Skipping point {quant_point}")
+
+        return float_name2quant_name
+
+    def get_quantized_graph(self,
+                            quantized_model: Model,
+                            repr_dataset: Callable = None):
+        """
+        Get a graph representation of the quantized model.
+
+        Args:
+            quantized_model: The quantized model.
+            repr_dataset: Representative dataset to use during the graph building (not used in Keras).
+
+        Returns:
+            Graph representation of the quantized model.
+        """
+        quant_graph = model_reader(quantized_model)
+        return quant_graph
+
+
+    def get_quant_graph_with_metrics(self,
+                                     quantized_model: keras.Model,
+                                     collected_data: Dict[str, Any],
+                                     xquant_config: XQuantConfig,
+                                     repr_dataset: Callable = None) -> Graph:
+        """
+        Generate the quantized graph with associated metrics.
```

Comment:
there is some overloading on the term "graph" in this module which I find confusing.
maybe emphasize that this graph is built solely for the presentation of the results on tensorboard.